### PR TITLE
Li/patch search method

### PR DESCRIPTION
**Pull request by wanliAlex**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
bug fix

* **What is the current behavior?** (You can also link to an open issue here)
we fix two bugs in this PR:
1. When an invalid search_method is received, Marqo is returning a 500 error instead of a 400 error
2. The 422 error does not contain `code` and `type` inside the response body

* **What is the new behavior (if this is a feature change)?**
1. An invalid search_method will get a correct 400 error now
2. The 422 error now correctly contain `code` and `type` inside the response body

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
yes

* **Related Python client changes** (link commit/PR here)
no

* **Related documentation changes** (link commit/PR here)
no

* **Other information**:
no

* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: tests/tensor_search/integ_tests/test_search_combined.py [hunk 1]
```diff
@@ -836,3 +838,34 @@ def test_get_lexical_search_term(self):
...
+        """Test that the ValidationError is raised when an incorrect search method is provided."""
+        invalid_search_methods = [
+            ("", "Empty string"),
+            (1, "Integer"),
+            ([], "List"),
+            ({"searchMethod": "LEXICAL"}, "Dictionary"),
+        ]
+        for search_method, search_method_type in invalid_search_methods:
+            with self.subTest(search_method_type=search_method_type):
+                with self.assertRaises(ValidationError):
```
Comment (papa99do) [inline tests/tensor_search/integ_tests/test_search_combined.py]: minor: check the error message starts with `Invalid search method`

**Review by papa99do — DISMISSED** [review]
LGTM

File: tests/tensor_search/test_api_exception_handler.py [hunk 2]
```diff
@@ -100,3 +107,29 @@ def test_base_exception_handler_unhandled_error(self, mock_api_exception_handler
...
 
+    async def test_validation_exception_handler_CorrectResponseBody(self):
+        """Ensure that the validation exception handler returns the correct response body when 422 is raised."""
+        # Create a mock request
+        mock_request = MagicMock(spec=Request)
+        # Manually create a RequestValidationError
+        errors = [
+            {
+                "loc": ("body", "name"),
+                "msg": "ensure this value has at most 10 characters",
```
Comment (papa99do) [inline tests/tensor_search/test_api_exception_handler.py:118]: Is this message generated by pydantic or is it controlled by us? I was wondering if we should only return this message to the end user as detail. But it lacks the property information.

**Review by papa99do — APPROVED** [review]

File: tests/tensor_search/test_api_exception_handler.py (hunk 2 above)
Comment (wanliAlex) [reply tests/tensor_search/test_api_exception_handler.py:118]: In this test, it is just a manually made one. We always returned the details before.

**Review by farshidz — APPROVED** [review]

**Comment by wanliAlex** [comment]
Regarding the error handler change, here is the comparison:
```mq.index("test-lexical").search("test", searchable_attributes={}, search_method="LEXICAL")```
Before this PR we get:
```MarqoWebError: MarqoWebError: MarqoWebError Error message: {"detail":[{"loc"["body","searchableAttributes"],"msg":"value is not a valid list","type":"type_error.list"}]}status_code: 422, type: unhandled_error_type, code: unhandled_error, link:```

After this PR we get:
```MarqoWebError: MarqoWebError: MarqoWebError Error message: {'detail': [{'loc': ['body', 'searchableAttributes'],'msg': 'value is not a valid list', 'type': 'type_error.list'}], 'code': 'unprocessable_entity', 'type': 'invalid_request', 'link': ''}status_code: 422, type: invalid_request, code: unprocessable_entity, link:```

**Pull request merged**